### ISSUE TITLE
How to compare hierarchical and flat clustering approaches when data is unlabelled?

### ISSUE THREAD

**Issue by xiangyum** [Computational-Content-Analysis-2018/Content-Analysis#2]
The notebooks tell me to do a silhouette analysis. This is trivial for the flat clustering given the helper function provided, but I wasn't sure how to do a silhouette analysis of the hierarchical clustering. Any help or hint would be appreciated. Thank you.

**Comment by reidmcy** [comment]
You need to change how the clusters are created in the `plotSilhouette` function, the reference code uses kmeans:
```
clusterer = sklearn.cluster.KMeans(n_clusters=n_clusters, random_state=10)
cluster_labels = clusterer.fit_predict(X)
```

You want to switch it to the ward hierarchical clusters for generating the labels. Consider these lines:
```
linkage_matrix = scipy.cluster.hierarchy.ward(X)
cluster_labels = scipy.cluster.hierarchy.fcluster(linkage_matrix, n_clusters, 'maxclust')
```
